AssertionError: Assignment target had type <class '_ast.Starred'>.

live-py-mode fails with the following error:

    ----------------------------------------------------------------------------------------------------------------- | 
    Traceback (most recent call last):                                                                                | 
      File "/usr/lib/python3.6/ast.py", line 253, in visit                                                            | 
        return visitor(node)                                                                                          | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 489, in visit_Module             | 
        new_node = self.generic_visit(node)                                                                           | 
      File "/usr/lib/python3.6/ast.py", line 308, in generic_visit                                                    | 
        value = self.visit(value)                                                                                     | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 72, in visit                     | 
        new_node = super(Tracer, self).visit(node)                                                                    | 
      File "/usr/lib/python3.6/ast.py", line 253, in visit                                                            | 
        return visitor(node)                                                                                          | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 307, in visit_Assign             | 
        existing_node.targets)                                                                                        | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 651, in _wrap_assignment_targets | 
        format_text = self._wrap_assignment_target(target)                                                            | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 636, in _wrap_assignment_target  | 
        return '({})'.format(', '.join(target_names))                                                                 | 
      File "/home/sasi/.emacs.d/elpa/live-py-mode-20170326.934/code_tracer.py", line 638, in _wrap_assignment_target  | 
        assert isinstance(target, Attribute), assert_message                                                          | 
    AssertionError: Assignment target had type <class '_ast.Starred'>.                                                | 
    ----------------------------------------------------------------------------------------------------------------- | 

upon encountering expressions of the type:

    *a, x = [0, 1, 2]

This happened both in an older version and the latest version from the Emacs repository.

--- a/plugin/PySrc/code_tracer.py
+++ b/plugin/PySrc/code_tracer.py
@@ -634,6 +634,8 @@ class Tracer(NodeTransformer):
         if isinstance(target, Tuple) or isinstance(target, List):
             target_names = map(self._wrap_assignment_target, target.elts)
             return '({})'.format(', '.join(target_names))
+        if isinstance(target, Starred):
+            return '*{}'.format(target.value.id)
         assert_message = 'Assignment target had type {}.'.format(type(target))
         assert isinstance(target, Attribute), assert_message
         names = self._get_attribute_names(target)

--- a/plugin/PySrc/code_tracer.py
+++ b/plugin/PySrc/code_tracer.py
@@ -634,7 +634,7 @@ class Tracer(NodeTransformer):
         if isinstance(target, Tuple) or isinstance(target, List):
             target_names = map(self._wrap_assignment_target, target.elts)
             return '({})'.format(', '.join(target_names))
-        if isinstance(target, Starred):
+        if Starred is not None and isinstance(target, Starred):
             return '*{}'.format(target.value.id)
         assert_message = 'Assignment target had type {}.'.format(type(target))
         assert isinstance(target, Attribute), assert_message

--- a/test/PySrc/tests/code_tracer_test.py
+++ b/test/PySrc/tests/code_tracer_test.py
@@ -1168,6 +1168,23 @@ a = [1, 2, 3]
         # VERIFY
         self.assertReportEqual(expected_report, report)
 
+    @skipIf(sys.version_info < (3, 0),
+            'starred assignment not supported before Python 3.0.')
+    def test_assign_starred(self):
+        # SETUP
+        code = """\
+*a, b = (2, 3)
+"""
+        expected_report = """\
+(*a, b) = (2, 3) """
+        tracer = CodeTracer()
+
+        # EXEC
+        report = tracer.trace_code(code)
+
+        # VERIFY
+        self.assertReportEqual(expected_report, report)
+
     def test_method_of_anonymous_object(self):
         # SETUP
         code = """\

